### PR TITLE
issue/1180: Use core ItemsQuestionModel

### PR DESCRIPTION
**Pull request by oliverfoster**
[#1180](https://github.com/adaptlearning/adapt_framework/issues/1180)
* Moved to use ItemsQuestionModel
* Centralised DOM updates into two functions, Selection, Marking
* Removed redundant functions

### CODE REVIEW

File: js/adapt-contrib-mcq.js [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
     './mcqView',
-    './mcqModel'
-], function(Adapt, McqView, McqModel) {
+    'core/js/models/itemsQuestionModel'
+], function(Adapt, McqView, ItemsQuestionModel) {
 
     return Adapt.register("mcq", {
         view: McqView,
-        model: McqModel
+        model: ItemsQuestionModel.extend({})
```
Comment (moloko) [inline js/adapt-contrib-mcq.js]: out of curiosity, why `.extend({})`?

Comment (oliverfoster) [reply js/adapt-contrib-mcq.js]: It creates a new entry in the inheritance chain. 
Even though we're not explicitly defining any new behaviour for this component type, if someone wanted to add behaviour to the mcq model only they can at this point of the chain.
If we didn't extend here, you'd only have the choice to add functions to the ItemsQuestionModel globally or the Mcq model instance.

Comment (moloko) [reply js/adapt-contrib-mcq.js]: OK makes sense, could you add a brief comment to that effect?

Comment (oliverfoster) [reply js/adapt-contrib-mcq.js]: Sure

File: js/mcqView.js [hunk 2]
```diff
@@ -6,87 +6,71 @@ define([
...
+        isCorrectAnswerShown: false,
+
+        initialize: function() {
+            QuestionView.prototype.initialize.apply(this, arguments);
+            this.update = _.debounce(this.update.bind(this), 1);
+            this.listenTo(this.model, {
+                "change:_isEnabled change:_isComplete change:_isSubmitted": this.update
+            });
+            this.listenTo(this.model.getChildren(), {
+                "change:_isActive": this.update
```
[the remark this answers is not in the document]
Comment (oliverfoster) [reply js/mcqView.js]: updateMarking will also remove marking if needed. Which is a bit necessary.

**Pull request merged**